### PR TITLE
Laboratory work #2, Khristine Tarasenko - 23FPL2

### PR DESCRIPTION
**Pull request by KhristineTarasenko**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -60,6 +60,17 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(documents, list) or not all(isinstance(doc, list) for doc in documents):
+        return None
+
+    unique_tokens = set()
+
+    for document in documents:
+        if any(not isinstance(token, str) for token in document):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:69]: check it earlier

**Review by EIzharikov — COMMENTED** [review]
You've made some progress, but it's still progress! Make CI green and fix&answer all my comments. Hurry up!

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -92,6 +163,20 @@ def calculate_tf_idf(tf: dict[str, float], idf: dict[str, float]) -> dict[str, f
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(tf, dict) or any(
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:166]: join checks. The same for other functions

File: lab_2_retrieval_w_bm25/settings.json [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 6
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/settings.json:2]: nice aim! maybe 8??

File: lab_2_retrieval_w_bm25/start.py [hunk 4]
```diff
@@ -26,7 +29,7 @@ def main() -> None:
             documents.append(file.read())
     with open("assets/stopwords.txt", "r", encoding="utf-8") as file:
         stopwords = file.read().split("\n")
-    result = None
+    result = tokenize(documents[0])
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/start.py:32]: show here the full presentation of your work

**Review by EIzharikov — COMMENTED** [review]
Make CI green, fix&answer all my coments and be ready for the next review. Good luck!

**Review by EIzharikov — COMMENTED** [review]
So, there are a lot of unanswered questions, it means that you will answer in Friday to me. Make CI green, answer&fix all my comments and be ready for Friday. Good luck.

File: lab_2_retrieval_w_bm25/main.py [hunk 5]
```diff
@@ -18,6 +20,9 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    return re.findall(r'\b[^\d\W]+\b', text.lower())
```
[the remark this answers is not in the document]
Comment (KhristineTarasenko) [reply lab_2_retrieval_w_bm25/main.py]: it didnt work properly, so I changed it

File: lab_2_retrieval_w_bm25/main.py (hunk 1 above)
Comment (KhristineTarasenko) [reply lab_2_retrieval_w_bm25/main.py:74]: done

File: lab_2_retrieval_w_bm25/start.py (hunk 4 above)
Comment (KhristineTarasenko) [reply lab_2_retrieval_w_bm25/start.py]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (KhristineTarasenko) [reply lab_2_retrieval_w_bm25/main.py:169]: done

**Review by EIzharikov — COMMENTED** [review]
Be ready for Friday, good luck!

**Pull request closed without merging**